Multi-index flush_cache is broken.

**Steps to reproduce**
- create contract with multi-index
- add some contract action with calling of `multi_index::flush_cache()`

**Actual result**
```
In file included from /usr/local/cyberway.cdt/bin/../include/eosiolib/contracts/eosio/eosio.hpp:8:
/usr/local/cyberway.cdt/bin/../include/eosiolib/contracts/eosio/multi_index.hpp:585:20: error: member reference type 'std::__1::pair<const unsigned long long, eosio::intrusive_ptr<eosio::multi_index_item<cyber::stake::structures::agent,
      eosio::multi_index<14289082693702709760, cyber::stake::structures::agent, eosio::indexed_by<3680915036248735744, eosio::const_mem_fun<cyber::stake::structures::agent, unsigned long long,
      &cyber::stake::structures::agent::primary_key> >, eosio::indexed_by<4584856835198025728, eosio::const_mem_fun<cyber::stake::structures::agent, std::__1::tuple<eosio::symbol_code, eosio::name>,
      &cyber::stake::structures::agent::by_key> > > > > >' is not a pointer; did you mean to use '.'?
                itr->second->deleted_ = true;
```

**Expected result**
No errors

--- a/libraries/eosiolib/contracts/eosio/multi_index.hpp
+++ b/libraries/eosiolib/contracts/eosio/multi_index.hpp
@@ -581,8 +581,8 @@ private:
             }
             vector.clear();
 
-            for (auto& itr: map) {
-                itr->second->deleted_ = true;
+            for (auto& itm: map) {
+                itm.second->deleted_ = true;
             }
             map.clear();
         }

